mantidproject/mantid
# Re-calculate duration in split-out workspaces from FilterEvents

**Description of work.**

Re-calculate the duration, as a property in run, of each output (split) workspace.

**To test:**

1. Review the expanded unit test
1. Load a workspace, generate splitter manually with known duration, call FilterEvents and then check the sample log 'duration' from all the output workspaces.

Fixes #27677 .




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
